pyomo/contrib/doe/utils.py

```diff
@@ -0,0 +1,97 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright (c) 2008-2024
+#  National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#
+#  Pyomo.DoE was produced under the Department of Energy Carbon Capture Simulation
+#  Initiative (CCSI), and is copyright (c) 2022 by the software owners:
+#  TRIAD National Security, LLC., Lawrence Livermore National Security, LLC.,
+#  Lawrence Berkeley National Laboratory, Pacific Northwest National Laboratory,
+#  Battelle Memorial Institute, University of Notre Dame,
+#  The University of Pittsburgh, The University of Texas at Austin,
+#  University of Toledo, West Virginia University, et al. All rights reserved.
+#
+#  NOTICE. This Software was developed under funding from the
+#  U.S. Department of Energy and the U.S. Government consequently retains
+#  certain rights. As such, the U.S. Government has been granted for itself
+#  and others acting on its behalf a paid-up, nonexclusive, irrevocable,
+#  worldwide license in the Software to reproduce, distribute copies to the
+#  public, prepare derivative works, and perform publicly and display
+#  publicly, and to permit other to do so.
+#  ___________________________________________________________________________
+
+import pyomo.environ as pyo
+
+from pyomo.common.dependencies import numpy as np, numpy_available
+
+from pyomo.core.base.param import ParamData
+from pyomo.core.base.var import VarData
+
+
+# Rescale FIM (a scaling function to help rescale FIM from parameter values)
+def rescale_FIM(FIM, param_vals):
+    """
+    Rescales the FIM based on the input and parameter vals.
+    It is assumed the parameter vals align with the FIM
+    dimensions such that (1, i) corresponds to the i-th
+    column or row of the FIM.
+
+    Parameters
+    ----------
+    FIM: 2D numpy array to be scaled
+    param_vals: scaling factors for the parameters
+
+    """
+    if isinstance(param_vals, list):
+        param_vals = np.array([param_vals])
+    elif isinstance(param_vals, np.ndarray):
+        if len(param_vals.shape) > 2 or (
+            (len(param_vals.shape) == 2) and (param_vals.shape[0] != 1)
+        ):
+            raise ValueError(
+                "param_vals should be a vector of dimensions (1, n_params). The shape you provided is {}.".format(
+                    param_vals.shape
+                )
+            )
+        if len(param_vals.shape) == 1:
+            param_vals = np.array([param_vals])
+    scaling_mat = (1 / param_vals).transpose().dot((1 / param_vals))
+    scaled_FIM = np.multiply(FIM, scaling_mat)
+    return scaled_FIM
+
+
+def get_parameters_from_suffix(suffix, fix_vars=False):
```

Comment:
Is this utility used anywhere? I couldn't find any references to it in this PR.

Reply:
This was going to be used with parmest's ``param_to_var`` functionality but since the method is still string based. This will be removed and added to the immediate "To-Do's".